source/Tool/GeneratorsImpl.cpp

```diff
@@ -73,17 +74,13 @@ insert(
 
 //------------------------------------------------
 
-GeneratorsImpl&
-getGeneratorsImpl() noexcept
-{
-    static GeneratorsImpl impl;
-    return impl;
-}
-
-Generators const&
+Generators &
 getGenerators() noexcept
 {
-    return getGeneratorsImpl();
+    auto & gens = toolWorld().generators;
+    if (gens == nullptr)
+        gens = std::make_unique<GeneratorsImpl>();
```

Comment:
why not do this in `ToolWorld::ToolWorld`?

Reply:
I don't like this. Checking every time doesn't really make sense once it is already created. And I don't like that the lifetime is somewhat indeterminate - it depends on when `getGeneratorsImpl` is called. This is not ideal. Instead, we should do it in the constructor as I wrote it, because then the beginning of the lifetime is crystal clear to the reader.